flake/dev/default.nix

```diff
@@ -0,0 +1,49 @@
+{ inputs, ... }:
+{
+  imports = [
+    inputs.treefmt-nix.flakeModule
+    ./dev-shell.nix
+    ./nixpkgs-rev.nix
+    ./packages.nix
+    ./pre-commit.nix
+  ];
+
+  perSystem = {
+    treefmt = {
```

Comment:
This feels like a good opportunity to move the treefmt config to a dedicated module (e.g. `./treefmt.nix` or `./formatting.nix`)